File: Semestrul 4/MPP/Proiecte/c#_gui/domain/Entity.cs
namespace csharp_avalonia.domain;

public class Entity<T>
{
    protected T id;

    public T Id
    {
        get => id;
        set => id = value;
    }

    public override bool Equals(object obj)
    {
        if (ReferenceEquals(this, obj))
            return true;
        if (obj == null || GetType() != obj.GetType())
            return false;

        return Id != null && Id.Equals(((Entity<T>)obj).Id);
    }

    public override int GetHashCode()
    {
        return id?.GetHashCode() ?? 0;
    }

    public override string ToString()
    {
        return $"Entity{{id={id}}}";
    }
}

File: Semestrul 4/MPP/Proiecte/java_gui/src/main/java/map/domain/Entity.java
package map.domain;

import java.io.Serializable;
import java.util.Objects;

public class Entity<ID> implements Serializable {

    protected ID id;

    public ID getId() {
        return id;
    }

    public void setId(ID id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;

        return getId().equals(((Entity) o).getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId());
    }

    @Override
    public String toString() {
        return "Entity{" + "id=" + id + '}';
    }

}